wgsl/index.bs

```diff
@@ -4149,16 +4148,24 @@ use the same memory.
 
 Expressions specify how values are computed.
 
-## `const` Expressions ## {#const-expr}
+## Early Evaluation Expressions ## {#early-eval-exprs}
+
+WGSL defines two types of expressions that can be evaluated before runtime:
+* [=const-expressions=], and
+* [=override-expressions=]
+
+### `const` Expressions ### {#const-expr}
 
 Expressions that are evaluated at [=shader module creation|shader-creation
 time=] are called <dfn noexport>const-expressions</dfn>.
 In order for an expression to be evaluated at shader-creation time all
-[=identifiers=] used by the expression must [=resolve=] to
-[=const-declarations=] or [=const-functions=].
+[=identifiers=] used by the expression must [=resolve=] to:
+* [=const-declarations=], or
+* [=const-functions=], or 
+* [=type aliases=], or 
+* [=structure=] names
 
-The types of `const` expressions can resolve to types that include
-[=abstract numeric types=].
+The types of `const` expressions can [=type rules|resolve=] to [=abstract types=].
```

Comment:
Since the "include" word is dropped, this seems like concrete types are not allowed.

There might not be a type category that fits.
The closest thing would be creation-fixed footprint, if we allowed those to have abstract members.

I *think* we can do this:
1. Say const-expr must have type that is creation-fixed footprint.  
2. Add a note saying that this may include abstract types.
2.  redefine creation-fixed footprint types to have abstract members.  Then that would fix the pending issue to allow arrays and structs for const-expr types.
3.  I think the other patch we would need is in the address space table to say that store type for workgroup must be *concrete* fixed-footprint instead of just fixed-footprint.


What's a little weird about this is we describe/define creation-fixed footprint as

> A plain type has a creation-fixed footprint if its size is fully determined at shader creation time.

But we'd be stretching the meaning of "size" quite far.  Is there a good tweak for that?
A _concretization_ of the type is deterministically sized....?

Reply:
Arrays will still not work with abstracts until we add the constructor that infers the component. The rest can happen though.

Reply:
Ah, I was originally concerned that this patch regresses functionality.
But concrete fixed-size array is still covered by 'fixed-footprint' because it falls under the idenity row.  So that makes sense now upon reflection.